morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -1,100 +1,237 @@
-import Ouroboros.Network.PeerSelection.LedgerPeers (TraceLedgerPeers)
-import Ouroboros.Network.TxSubmission.Inbound (TraceTxSubmissionInbound)
-import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound)
-import Prettyprinter (Doc, Pretty (..), list, viaShow, (<+>))
+import Ouroboros.Network.PeerSelection.LedgerPeers (AccPoolStake (AccPoolStake), IP (IPv4, IPv6), PoolStake (PoolStake), RelayAddress (RelayAddressAddr, RelayAddressDomain), TraceLedgerPeers (FetchingNewLedgerState, PickedPeer, PickedPeers))
+import Ouroboros.Network.PeerSelection.RootPeersDNS (Domain, DomainAddress (DomainAddress))
+import Ouroboros.Network.Point (WithOrigin (At, Origin))
+import Ouroboros.Network.Protocol.BlockFetch.Type (ClientHasAgency, Message, ServerHasAgency)
+import qualified Ouroboros.Network.Protocol.BlockFetch.Type as BlockFetch
+import qualified Ouroboros.Network.Protocol.ChainSync.Type as ChainSync
+import Ouroboros.Network.Protocol.Handshake.Type (Handshake)
+import qualified Ouroboros.Network.Protocol.Handshake.Type as Handshake
+import qualified Ouroboros.Network.Protocol.Trans.Hello.Type as Hello
+import qualified Ouroboros.Network.Protocol.TxSubmission.Type as TxSubmission
+import Ouroboros.Network.Snocket (FileDescriptor, LocalAddress (LocalAddress))
+import Ouroboros.Network.Subscription (ConnectResult (ConnectSuccess, ConnectSuccessLast, ConnectValencyExceeded))
+import Ouroboros.Network.TxSubmission.Inbound (ProcessedTxCount (ProcessedTxCount), TraceTxSubmissionInbound (TraceTxInboundCanRequestMoreTxs, TraceTxInboundCannotRequestMoreTxs, TraceTxInboundTerminated, TraceTxSubmissionCollected, TraceTxSubmissionProcessed))
+import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound (TraceControlMessage, TraceTxSubmissionOutboundRecvMsgRequestTxs, TraceTxSubmissionOutboundSendMsgReplyTxs))
+import Prettyprinter (Doc, Pretty (..), comma, concatWith, hsep, punctuate, viaShow, (<+>))
 
 class MPretty a where
   mpretty :: a -> Doc ann
 
-instance (HashAlgorithm h, BftCrypto c) => MPretty (MorphoBlock h c) where
-  mpretty MorphoBlock {..} =
-    "("
-      <> mpretty morphoPrev
-      <> "->"
-      <> mpretty morphoHeaderHash
-      <> ","
-      <> mpretty morphoSlotNo
-      <> ","
-      <> mpretty morphoTxs
+instance MPretty (HeaderHash blk) => MPretty (MorphoState blk) where
+  mpretty (MorphoState lastCkpt ckptAt votes tip) =
+    "MorphoState(morphoTip:"
+      <+> mpretty tip
+      <> ", lastCheckpoint:"
+      <+> mpretty lastCkpt
+      <> ", checkpointAt:"
+      <+> mpretty ckptAt
+      <> ", currentVotes:"
+      <+> mpretty (Map.elems votes)
       <> ")"
-    where
-      MorphoHeader {..} = morphoHeader
-      MorphoStdHeader {..} = morphoHeaderStd
-      MorphoBody {..} = morphoBody
 
-instance
-  (HashAlgorithm h, BftCrypto c) =>
-  MPretty (Header (MorphoBlock h c))
-  where
+instance MPretty PublicKey where
+  mpretty (PublicKey bytes) =
+    "PublicKey(" <> mpretty bytes <> ")"
```

Comment:
And here let's change this to just

```suggestion
    mpretty bytes
```

The fact that it's a `PublicKey` is always obvious from context.